test/test_change_stream.py

```diff
@@ -390,5 +504,188 @@ def test_read_concern(self):
+        db = client.get_database(scenario_def["database_name"])
+        cs_target = db.get_collection(scenario_def["collection_name"])
+    elif target == "database":
+        cs_target = client.get_database(scenario_def["database_name"])
+    elif target == "client":
+        cs_target = client
+    else:
+        raise ValueError("Invalid target in spec")
+
+    # Construct change stream kwargs dict
+    cs_pipeline = test["changeStreamPipeline"]
+    options = test["changeStreamOptions"]
+    cs_options = {}
+    for key, value in options.iteritems():
+        cs_options[camel_to_snake(key)] = value
+    
+    # Create and return change stream
+    with cs_target.watch(pipeline=cs_pipeline, **cs_options) as change_stream:
+        yield change_stream
+
+
+def run_operation(client, operation):
+    # Apply specified operations
+    opname = camel_to_snake(operation["name"])
+    arguments = operation["arguments"]
+    cmd = getattr(client.get_database(
+        operation["database"]).get_collection(
+        operation["collection"]), opname
+    )
+    return cmd(**arguments)
+
+
+def assert_dict_is_subset(superdict, subdict):
+    """Check that subdict is a subset of superdict."""
+    exempt_fields = ["documentKey", "_id"]
+    for key, value in subdict.iteritems():
+        if not superdict.has_key(key):
+            assert False
+        if isinstance(value, dict):
+            assert_dict_is_subset(superdict[key], value)
+            continue
+        if key in exempt_fields:
+            superdict[key] = "42"
+        assert superdict[key] == value
+
+
+def check_event(event, expectation_dict):
+    if event is None:
+        raise AssertionError
+    for key, value in expectation_dict.iteritems():
+        if isinstance(value, dict):
+            assert_dict_is_subset(
+                getattr(event, key), value
+            )
+        else:
+            assert getattr(event, key) == value
+
+
+def create_test(scenario_def, test):
+    def run_scenario(self):
+        # Set up
+        self.setUpCluster(scenario_def)
+        try:
+            with get_change_stream(
+                self.client, scenario_def, test
+            ) as change_stream:
+                for operation in test["operations"]:
+                    # Run specified operations
+                    run_operation(self.client, operation)
+                num_expected_changes = len(test["result"]["success"])
+                changes = [
+                    change_stream.next() for _ in range(num_expected_changes)
+                ]
+
+        except OperationFailure as exc:
+            if test["result"].get("error") is None:
+                self.fail("Encountered unexpected error")
+            expected_code = test["result"]["error"]["code"]
+            self.assertEqual(exc.code, expected_code)
+
+        else:
+            # Check for expected output from change streams
+            for change, expected_changes in zip(changes, test["result"]["success"]):
```

Comment:
After this assert that `changes` and `test["result"]` have the same length. (after so that you get the more informative error first)